### DISCUSSION TITLE
x-model doesn't work in the CSP build

### DISCUSSION THREAD

**Discussion by akhan619** [alpinejs/alpine#1847] [7. Bugs (archived)] [answered]
# #Summary
x-model is not reactive when used in the CSP build.

## Test
I cloned the repo and did both ```npm install && npm run build```. No issues there. From there included the ```module.esm.js``` from the csp folder in my app.js and set it up as follows
```
import Alpine from './alpinejs/csp/module.esm'

window.Alpine = Alpine

document.addEventListener('alpine:init', () => {
    Alpine.data('data', () => ({
        message: ''
    }))
})

window.Alpine.start()
```

Tested it with the example in the documentation
```
    <div x-data="data">
        <input type="text" x-model="message">
        <span x-text="message"></span>
    </div>
```

This is not working. As a test used the normal build i.e. non-csp and everything works as expected. Side-note: Any initial value set for ```message``` does reflect in both the input and span. However, they aren't reactive.

## Questions
- Is the CSP build complete? 
- Am I making a mistake in assuming that the we can bind to the Alpine.data() object as I did?

Any feedback is appreciated.

**Comment by SimoTod** [comment, marked as answer]
The CSP plugin hasn't been officially released yet so it's probably not complete.

**Reply by akhan619** [reply]
That would make sense as to why some directives work and others don't. But it is confusing that there is no warning about it not being complete.

The official docs has a CSP section in advanced and talks about how to use it. And there are instructions on the repo to build it. So one can assume it is supported and complete. 

Still thanks for the clarification @SimoTod. Hope to see the CSP soon as it should be something that makes AlpineJs even more attractive, if not already.

**Reply by sbosell** [reply]
Unfortunately, x-model still doesn't work.
